backend/src/main/java/com/zeros/backend/api/RestaurantAffiliateNetworkResource.java

```diff
@@ -0,0 +1,57 @@
+package com.zeros.backend.api;
+
+import com.zeros.backend.exceptions.AlreadyAffiliatedRestaurantException;
+import com.zeros.backend.exceptions.NullQrException;
+import com.zeros.backend.exceptions.RestaurantNotFoundException;
+import com.zeros.backend.models.RestaurantAffiliate;
+import com.zeros.backend.services.RestaurantAffiliateNetworwk;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/v1")
+public class RestaurantAffiliateNetworkResource {
+
+    @Autowired
+    private RestaurantAffiliateNetworwk restaurantAffiliateNetworwk;
+
+    @PostMapping("/restaurant/affiliate")
+    public void affiliate(@RequestBody RestaurantAffiliateDto restaurantAffiliate) throws RestaurantNotFoundException {
+        restaurantAffiliateNetworwk.affiliate(restaurantAffiliate.getRestaurantId(), restaurantAffiliate.getQrCode());
+    }
+
+    @ExceptionHandler(RestaurantNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public void handleResourceDoesNotExistException() {
+    }
+
+    @ExceptionHandler(AlreadyAffiliatedRestaurantException.class)
+    @ResponseStatus(HttpStatus.CONFLICT)
+    public void handleResourceAlreadyExistException() {
+
+    }
+
+    @ExceptionHandler(NullQrException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public void handleBadRequestException() {
+    }
+
+
+    static class RestaurantAffiliateDto {
+
+        private Long restaurantId;
+        private Long qrCode;
+
```

Comment:
@JumperSpirit  , i don't think that when we want to subscribe a restaurant into our network, we ask for an id or qr code ;
we need only information about the restaurant like name , address

Reply:
I think these informations should be in database before 
restaurant can accepte or not to be affiliate